src/mbgl/style/layer.cpp

```diff
@@ -143,6 +143,35 @@ void Layer::setObserver(LayerObserver* observer_) {
     observer = observer_ ? observer_ : &nullObserver;
 }
 
+optional<conversion::Error> Layer::setProperty(const std::string& name, const conversion::Convertible& value) {
+    using namespace conversion;
+    optional<Error> error = setPropertyInternal(name, value);
```

Comment:
nit: maybe worth switching blocks, common first with fallback to layer specific.

Reply:
layer-specific shall be faster (have eternal map there)